### ISSUE TITLE
Disabling eslint rule for react/forbid-foreign-prop-types does not work on all references

### ISSUE THREAD

**Issue by noratarano** [yannickcr/eslint-plugin-react#2328]
### Summary of the issue
Here's a snippet from the code in our repo for prop types hoisting in an HOC:
 
```tsx
function withBreakpoint<C extends React.ComponentType<any>>(WrappedComponent: C) {
  // ...

  // eslint-disable-next-line react/forbid-foreign-prop-types
  if (WrappedComponent.propTypes) {
    // eslint-disable-next-line react/forbid-foreign-prop-types
    const { currentBreakpoint, breakpoints, ...restPropTypes } = WrappedComponent.propTypes;
    WithBreakpoint.propTypes = restPropTypes as PropTypes.ValidationMap<Props>;
  }

  // ...
}
```

Even though linting on the the first reference of `WrappedComponent.propTypes` is disabled, it fails linting.

**Screenshot:**

![Screen Shot 2019-06-24 at 7 38 32 PM](https://user-images.githubusercontent.com/1694863/60065151-b4eb1e80-96b7-11e9-81fc-4183307da781.png)

**Result in CI:**
```console
133:24  error    Using propTypes from another component is not safe because they may be removed in production builds  react/forbid-foreign-prop-types
```

### Attempted solutions

**❌ Didn't work**
- Surrounded the statement with `/* eslint-disable react/forbid-foreign-prop-types */` and `/* eslint-enable react/forbid-foreign-prop-types */`
- `// eslint-disable-next-line` to disable all rules for that line

**✅ Worked**
- Added `/* eslint-disable react/forbid-foreign-prop-types */` at the top of the file

cc @ljharb

**Comment by ljharb** [comment]
That should certainly work. What's your eslint config and versions?

**Comment by yannickcr** [comment]
I was not be able to reproduce the issue on the above snipped using `eslint@6.01` and `eslint-plugin-react@7.14.2`.

**Comment by ljharb** [comment]
Closing pending confirmation of version numbers used.

**Comment by ljharb** [comment]
This was happening with eslint 5; reopening.

**Comment by yannickcr** [comment]
Was not able to reproduce it with ESLint@5 either, do you have more information @ljharb?

**Comment by ljharb** [comment]
Closing.

Happy to reopen and fix ASAP with a repro repo, or a failing test, but it's unlikely this continues to be a problem.
